From 6c5515de615dde5d713e59fa9e592f231fa066d2 Mon Sep 17 00:00:00 2001
From: Joseph Guo <qijian.guo@nxp.com>
Date: Thu, 19 Jun 2025 14:41:24 +0900
Subject: [PATCH 24/56] LF-15689: arm64: dts: imx93-11x11-frdm: add DT to
 support Tianma TM050RDH03 panel

Add Tianma TM050RDH03 5.0" WVGA TFT LCD panel support on FRDM-IMX93 platform.

Signed-off-by: Joseph Guo <qijian.guo@nxp.com>
Acked-by: Liu Ying <victor.liu@nxp.com>
---
 arch/arm64/boot/dts/freescale/Makefile        |   3 +
 .../imx93-11x11-frdm-tianma-wvga-panel.dtso   | 185 ++++++++++++++++++
 2 files changed, 188 insertions(+)
 create mode 100644 arch/arm64/boot/dts/freescale/imx93-11x11-frdm-tianma-wvga-panel.dtso

diff --git a/arch/arm64/boot/dts/freescale/Makefile b/arch/arm64/boot/dts/freescale/Makefile
index 28d37a25c6e8..8b0d2b673982 100644
--- a/arch/arm64/boot/dts/freescale/Makefile
+++ b/arch/arm64/boot/dts/freescale/Makefile
@@ -500,6 +500,9 @@ dtb-$(CONFIG_ARCH_MXC) += imx93-11x11-frdm.dtb imx93-11x11-frdm-8mic-reve.dtb \
 imx93-11x11-frdm-ov5640-dtbs := imx93-11x11-frdm.dtb imx93-11x11-frdm-ov5640.dtbo
 dtb-$(CONFIG_ARCH_MXC) += imx93-11x11-frdm-ov5640.dtb
 
+imx93-11x11-frdm-tianma-wvga-panel-dtbs := imx93-11x11-frdm.dtb imx93-11x11-frdm-tianma-wvga-panel.dtbo
+dtb-$(CONFIG_ARCH_MXC) += imx93-11x11-frdm-tianma-wvga-panel.dtb
+
 imx93-11x11-frdm-waveshare-7inch-c-panel-dtbs := imx93-11x11-frdm.dtb imx93-11x11-frdm-waveshare-7inch-c-panel.dtbo
 dtb-$(CONFIG_ARCH_MXC) += imx93-11x11-frdm-waveshare-7inch-c-panel.dtb
 
diff --git a/arch/arm64/boot/dts/freescale/imx93-11x11-frdm-tianma-wvga-panel.dtso b/arch/arm64/boot/dts/freescale/imx93-11x11-frdm-tianma-wvga-panel.dtso
new file mode 100644
index 000000000000..3dde9fe3bebc
--- /dev/null
+++ b/arch/arm64/boot/dts/freescale/imx93-11x11-frdm-tianma-wvga-panel.dtso
@@ -0,0 +1,185 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
+/*
+ * Copyright 2025 NXP
+ */
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/pwm/pwm.h>
+#include "imx93-pinfunc.h"
+
+/dts-v1/;
+/plugin/;
+
+&{/} {
+	backlight: backlight {
+		compatible = "pwm-backlight";
+		pwms = <&tpm5 3 40000 PWM_POLARITY_INVERTED>;	/* 25KHz PWM */
+		brightness-levels = <0 100>;
+		num-interpolated-steps = <100>;
+		default-brightness-level = <80>;
+		power-supply = <&reg_vexp_5v>;
+	};
+
+	panel {
+		compatible = "tianma,tm050rdh03";
+		standby-gpios = <&gpio2 27 GPIO_ACTIVE_LOW>;	/* STBYB */
+		backlight = <&backlight>;
+		power-supply = <&reg_vexp_3v3>;
+
+		port {
+			panel_in: endpoint {
+				remote-endpoint = <&display_out>;
+			};
+		};
+	};
+};
+
+/* pin conflicts */
+&flexcan2 {
+	status = "disabled";
+};
+
+&gpio2 {
+	/* avoid power leakage */
+	disp-clk-hog {
+		gpio-hog;
+		gpios = <0 GPIO_ACTIVE_HIGH>;
+		output-low;
+	};
+
+	disp-de-hog {
+		gpio-hog;
+		gpios = <1 GPIO_ACTIVE_HIGH>;
+		output-low;
+	};
+
+	disp-vsync-hog {
+		gpio-hog;
+		gpios = <2 GPIO_ACTIVE_HIGH>;
+		output-low;
+	};
+
+	disp-hsync-hog {
+		gpio-hog;
+		gpios = <3 GPIO_ACTIVE_HIGH>;
+		output-low;
+	};
+};
+
+&iomuxc {
+	pinctrl_lcdif: lcdifgrp {
+		fsl,pins = <
+			MX93_PAD_GPIO_IO00__MEDIAMIX_DISP_CLK		0x31e
+			MX93_PAD_GPIO_IO01__MEDIAMIX_DISP_DE		0x31e
+			MX93_PAD_GPIO_IO02__MEDIAMIX_DISP_VSYNC		0x31e
+			MX93_PAD_GPIO_IO03__MEDIAMIX_DISP_HSYNC		0x31e
+			MX93_PAD_GPIO_IO04__MEDIAMIX_DISP_DATA00	0x31e
+			MX93_PAD_GPIO_IO05__MEDIAMIX_DISP_DATA01	0x31e
+			MX93_PAD_GPIO_IO06__MEDIAMIX_DISP_DATA02	0x31e
+			MX93_PAD_GPIO_IO07__MEDIAMIX_DISP_DATA03	0x31e
+			MX93_PAD_GPIO_IO08__MEDIAMIX_DISP_DATA04	0x31e
+			MX93_PAD_GPIO_IO09__MEDIAMIX_DISP_DATA05	0x31e
+			MX93_PAD_GPIO_IO10__MEDIAMIX_DISP_DATA06	0x31e
+			MX93_PAD_GPIO_IO11__MEDIAMIX_DISP_DATA07	0x31e
+			MX93_PAD_GPIO_IO12__MEDIAMIX_DISP_DATA08	0x31e
+			MX93_PAD_GPIO_IO13__MEDIAMIX_DISP_DATA09	0x31e
+			MX93_PAD_GPIO_IO14__MEDIAMIX_DISP_DATA10	0x31e
+			MX93_PAD_GPIO_IO15__MEDIAMIX_DISP_DATA11	0x31e
+			MX93_PAD_GPIO_IO16__MEDIAMIX_DISP_DATA12	0x31e
+			MX93_PAD_GPIO_IO17__MEDIAMIX_DISP_DATA13	0x31e
+			MX93_PAD_GPIO_IO18__MEDIAMIX_DISP_DATA14	0x31e
+			MX93_PAD_GPIO_IO19__MEDIAMIX_DISP_DATA15	0x31e
+			MX93_PAD_GPIO_IO20__MEDIAMIX_DISP_DATA16	0x31e
+			MX93_PAD_GPIO_IO21__MEDIAMIX_DISP_DATA17	0x31e
+			MX93_PAD_GPIO_IO27__GPIO2_IO27			0x31e
+		>;
+	};
+
+	pinctrl_lcdif_gpio: lcdifgpiogrp {
+		fsl,pins = <
+			MX93_PAD_GPIO_IO00__GPIO2_IO00			0x51e
+			MX93_PAD_GPIO_IO01__GPIO2_IO01			0x51e
+			MX93_PAD_GPIO_IO02__GPIO2_IO02			0x51e
+			MX93_PAD_GPIO_IO03__GPIO2_IO03			0x51e
+			MX93_PAD_GPIO_IO04__GPIO2_IO04			0x51e
+			MX93_PAD_GPIO_IO05__GPIO2_IO05			0x51e
+			MX93_PAD_GPIO_IO06__GPIO2_IO06			0x51e
+			MX93_PAD_GPIO_IO07__GPIO2_IO07			0x51e
+			MX93_PAD_GPIO_IO08__GPIO2_IO08			0x51e
+			MX93_PAD_GPIO_IO09__GPIO2_IO09			0x51e
+			MX93_PAD_GPIO_IO10__GPIO2_IO10			0x51e
+			MX93_PAD_GPIO_IO11__GPIO2_IO11			0x51e
+			MX93_PAD_GPIO_IO12__GPIO2_IO12			0x51e
+			MX93_PAD_GPIO_IO13__GPIO2_IO13			0x51e
+			MX93_PAD_GPIO_IO14__GPIO2_IO14			0x51e
+			MX93_PAD_GPIO_IO15__GPIO2_IO15			0x51e
+			MX93_PAD_GPIO_IO16__GPIO2_IO16			0x51e
+			MX93_PAD_GPIO_IO17__GPIO2_IO17			0x51e
+			MX93_PAD_GPIO_IO18__GPIO2_IO18			0x51e
+			MX93_PAD_GPIO_IO19__GPIO2_IO19			0x51e
+			MX93_PAD_GPIO_IO20__GPIO2_IO20			0x51e
+			MX93_PAD_GPIO_IO21__GPIO2_IO21			0x51e
+			MX93_PAD_GPIO_IO27__GPIO2_IO27			0x51e
+		>;
+	};
+
+	pinctrl_tpm5_ch3: tpm5ch3grp {
+		fsl,pins = <
+			MX93_PAD_GPIO_IO26__TPM5_CH3			0x31e
+		>;
+	};
+};
+
+&lcdif {
+	assigned-clock-rates = <300000000>, <30000000>, <333333334>, <125000000>;
+	status = "okay";
+};
+
+&ldb {
+	status = "disabled";
+};
+
+&ldb_phy {
+	status = "disabled";
+};
+
+/* pin conflicts */
+&lpspi3 {
+	status = "disabled";
+};
+
+&lvds_bridge {
+	status = "disabled";
+};
+
+&parallel_disp_fmt {
+	pinctrl-names = "default", "gpio";
+	pinctrl-0 = <&pinctrl_lcdif>;
+	pinctrl-1 = <&pinctrl_lcdif_gpio>;
+	fsl,interface-pix-fmt = "rgb666";
+	status = "okay";
+
+	ports {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		port@1 {
+			reg = <1>;
+
+			display_out: endpoint {
+				remote-endpoint = <&panel_in>;
+			};
+		};
+	};
+};
+
+/* pin conflicts */
+&sai3 {
+	status = "disabled";
+};
+
+&tpm5 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_tpm5_ch3>;
+	status = "okay";
+};
-- 
2.34.1

